src/useradd.c

```diff
@@ -2210,6 +2211,15 @@ static void create_mail (void)
 			sprintf (file, "%s/%s/%s", prefix, spool, user_name);
 		else
 			sprintf (file, "%s/%s", spool, user_name);
+
+#ifdef WITH_SELINUX
+		if (set_selinux_file_context (spool) != 0) {
```

Comment:
why are you using the path of `spool` and not `file`?

Reply:
I agree that this may not be a good idea from a security point of view. So, I'm changing the fix to change directly the file.